### DISCUSSION TITLE
How to detect on the client side if a stream connection is broken

### DISCUSSION THREAD

**Discussion by vdejager** [projectdiscovery/httpx#1800] [Q&A]
We have a hikvision cam that we use for wildlife detection.
The system works by sending a GET request, upon which the camera returns a form-data response (mix of xml and binary data) for each event.
Time between events may take hours/days. In between, is there a way to detect if the stream is disconnected (with a callback or event hook maybe?) 
The idea is that if the connection is dropped, we can restart the stream (we have to use a flaky 4G connection and sometimes the signal drops out)

We use something like
```async with httpx.AsyncClient(auth=self._auth_method) as client:
            async with client.stream(
                method, full_url, timeout=timeout, **data
            ) as response:
                   #(process the response)....
                   # somewhere here we need to detect if the stream is disrupted

```

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
